File: client/src/dataModels/chiefComplaint.ts
import { BaseEntityModel } from "./baseEntityModel";

export class ChiefComplaint extends BaseEntityModel {
    Name: string;
    Title: string;
    CompanyId: string;

    constructor(id: string = "", isDelete: boolean = false) {
        super(id, isDelete);

        this.Name = "";
        this.Title = "";
        this.CompanyId = "";
    }

    createFromEntityModel(entity: any) {
        const chiefComplaint = new ChiefComplaint(entity.Id, !!entity.isDelete);

        chiefComplaint.Name = entity.Name;
        chiefComplaint.Title = entity.Title;
        chiefComplaint.CompanyId = entity.CompanyId;

        return chiefComplaint;
    }
}

File: medico/Domain/Medico.Domain/Models/ChiefComplaint.cs
using System;
using System.Collections.Generic;

namespace Medico.Domain.Models
{
    public class ChiefComplaint : Entity
    {
        public string Name { get; set; }

        public string Title { get; set; }

        public Guid CompanyId { get; set; }
        
        public Company Company { get; set; }

        public List<ChiefComplaintTemplate> ChiefComplaintTemplates { get; set; }

        public List<ChiefComplaintRelatedKeyword> ChiefComplaintsKeywords { get; set; }
    }
}
